### DISCUSSION TITLE
Idea: Add PWM cycle to dew-heater module

### DISCUSSION THREAD

**Discussion by lordwimsey** [AllskyTeam/allsky-modules#29] [Ideas]
Hi, 
first of all, thank you very much for programming and maintaining all these modules!
I just "finished" my first allsky-camera build and went for a dew heater right away. 

I used 6x 10 Ohm 1W resistors in series, so 60 Ohm together. With 12V supplied that would result in 200mA current (I = U/R) and therefore a power of 2.4W (P = U*I).
However, trying that out, I found the resistors got quite hot to the touch. Whether that is too hot, I can't tell- I have no experience.

That gave me the idea to have a load quotient or something (excuse my english...) that can switch on and off the heater in some PWM manner to reduce the effective heating output as an option in the settings. You could try to go with a percentage of the max. and see if it suffices. 
That way, we could save power and unnecessary hot environment for the camera.

Thanks for all of this and best regards,

Peter

By the way, the module documentation I found lacks some of the generated variables that I found in the Overlay Editor.

**Comment by Alex-developer** [comment]
Peter

Its a pleasure, Its a sideline for me as my main effort is developing Allsky itself. 

This is on my todo list to take a look at. The problem I need to get around is having PWM generated on a GPIO pin without having any code running. The heater module only runs periodically, defined by the settings in Allsky so currently the PWM signal would stop when the module stops.

I am sure I have seen some people use Hardware PWM on pi's but I have not looked into it yet

The module docs could do with an update !

Cheers

Alex

**Comment by Alex-developer** [comment]
Closing this as it will be handled under the issue https://github.com/AllskyTeam/allsky-modules/issues/41
